src/enlyze/client.py

```diff
@@ -150,6 +183,88 @@ def get_variables(
             for variable in self._get_variables(machine.uuid)
         ]
 
+    def _get_paginated_timeseries(
+        self,
+        *,
+        machine_uuid: str,
+        start: datetime,
+        end: datetime,
+        variables: Sequence[str],
+        resampling_interval: Optional[int],
+    ) -> Iterator[timeseries_api_models.TimeseriesData]:
+        params: dict[str, Any] = {
+            "appliance": machine_uuid,
+            "start_datetime": start.isoformat(),
+            "end_datetime": end.isoformat(),
+            "variables": ",".join(variables),
+        }
+
+        if resampling_interval:
+            params["resampling_interval"] = resampling_interval
+
+        return self._timeseries_api_client.get_paginated(
+            "timeseries", timeseries_api_models.TimeseriesData, params=params
+        )
+
+    def _get_timeseries(
+        self,
+        start: datetime,
+        end: datetime,
+        variables: Union[
+            Sequence[user_models.Variable],
+            Mapping[user_models.Variable, user_models.ResamplingMethod],
+        ],
+        resampling_interval: Optional[int] = None,
+    ) -> Optional[user_models.TimeseriesData]:
+        try:
+            variables_sequence, variables_query_parameter_list = (
+                _get_variables_sequence_and_query_parameter_list(
+                    variables, resampling_interval
+                )
+            )
+        except ValueError as e:
+            raise EnlyzeError from e
```

Comment:
> Shall we wrap the error message from ValueError into the EnlyzeError? Otherwise, the user would just see the EnlyzeError and needs to dig through the traceback.

That's not true

```python
>>> raise Exception from ValueError("Inner Error")
ValueError: Inner Error

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception
```

> the error message might not be visible at all if the exception is logged as a single line.

That's the service's problem IMO, anyways, the message will be on the first line.

Alas, I've changed this piece of code so that `_get_variables_sequence_and_query_parameter_list(...)` directly raises an `EnlyzeError` with the appropriate message because I don't see much in it raising a `ValueError` in this context.

Reply:
I've now reused the `ResamplingValidationError` that we already have as I think it makes more sense.

Reply:
> That's not true
> 
> ```python
> >>> raise Exception from ValueError("Inner Error")
> ValueError: Inner Error
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> Exception
> ```

That's an artifical example. If the traceback is very long, you wouldn't see on first sight that it's a chained exception without scrolling up. That's what I meant.

But doesn't matter anymore, you've changed it. Thanks :)